habitat/core/env.py

```diff
@@ -209,7 +209,12 @@ def _elapsed_seconds(self) -> float:
         return time.time() - self._episode_start_time
 
     def get_metrics(self) -> Metrics:
-        return self._task.measurements.get_metrics()
+        metrics = self._task.measurements.get_metrics()
+
+        if self._episode_over:
+            metrics.update(self._task.get_metrics_at_episode_end())
+
```

Comment:
I find this too awkward. Is it not possible to call the gfx code inside of a metric proper to avoid calling get_metrics_at_episode_end on the task?